src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.CompletionListUpdater.cs

```diff
@@ -47,7 +46,7 @@ private sealed class CompletionListUpdater
             private readonly bool _highlightMatchingPortions;
             private readonly bool _showCompletionItemFilters;
 
-            private readonly Func<ImmutableArray<(RoslynCompletionItem, PatternMatch?)>, string, ImmutableArray<RoslynCompletionItem>> _filterMethod;
+            private readonly Action<IEnumerable<(RoslynCompletionItem, PatternMatch?)>, string, IList<RoslynCompletionItem>> _filterMethod;
```

Comment:
don't love taking a collection type weakly as IEnumerable.  Can we tie it to a more strong interface which indicates the value will be realized and not lazy?

Reply:
Fixed